src/forest.rs

```diff
@@ -155,7 +158,10 @@ where
                 .flatten()
                 .cloned()
                 .map(move |i| {
-                    let (left, right, _) = node.range.split_at(i);
+                    let (left, right) = node
+                        .range
+                        .split_at::<Unknown>(/*i*/ unimplemented!())
```

Comment:
`self.possible_splits` will need to store trusted indices to be able to split here.